app/src/main/java/com/bogdan/codeforceswatcher/features/problems/ProblemsFragment.kt

```diff
@@ -1,126 +1,389 @@
+import android.view.ViewGroup
+import androidx.compose.animation.*
+import androidx.compose.animation.core.*
+import androidx.compose.foundation.Image
+import androidx.compose.ui.platform.ComposeView
 import androidx.fragment.app.Fragment
-import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
-import com.bogdan.codeforceswatcher.R
-import com.bogdan.codeforceswatcher.components.WebViewActivity
-import io.xorum.codeforceswatcher.features.problems.redux.ProblemsRequests
+import com.bogdan.codeforceswatcher.components.compose.theme.AlgoismeTheme
 import io.xorum.codeforceswatcher.features.problems.redux.ProblemsState
-import io.xorum.codeforceswatcher.redux.analyticsController
 import io.xorum.codeforceswatcher.redux.store
-import io.xorum.codeforceswatcher.util.AnalyticsEvents
-import kotlinx.android.synthetic.main.fragment_problems.*
 import tw.geothings.rekotlin.StoreSubscriber
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.dp
+import com.bogdan.codeforceswatcher.components.WebViewActivity
+import io.xorum.codeforceswatcher.util.AnalyticsEvents
+import com.bogdan.codeforceswatcher.R
+import com.google.accompanist.swiperefresh.SwipeRefresh
+import com.google.accompanist.swiperefresh.SwipeRefreshState
+import com.google.accompanist.swiperefresh.rememberSwipeRefreshState
+import io.xorum.codeforceswatcher.features.problems.redux.ProblemsRequests
+import io.xorum.codeforceswatcher.redux.analyticsController
 
-class ProblemsFragment : Fragment(), StoreSubscriber<ProblemsState>,
-    SwipeRefreshLayout.OnRefreshListener {
-
-    private lateinit var problemsAdapter: ProblemsAdapter
-    var searchView: SearchView? = null
+class ProblemsFragment : Fragment(), StoreSubscriber<ProblemsState> {
 
-    override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState)
-        setHasOptionsMenu(true)
-    }
+    private val problemsState: MutableState<List<UIModel>> = mutableStateOf(emptyList())
```

Comment:
I think that we should have a single `problemsState: MutableState<ProblemsState>` here, no need for UI models or  `SwipeRefreshState`. They add complexity and break the declarative approach a bit.